storage/test/transfer.test.js

```diff
@@ -0,0 +1,639 @@
+// Copyright 2016, Google, Inc.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+var proxyquire = require('proxyquire').noCallThru();
+var bucketName = 'foo';
+var jobName = 'transferJobs/123456789012345678';
+var transferOperationName = 'transferOperations/123456789012345678';
+
+function getSample () {
+  var transferJobMock = {};
+  var transferOperationMock = {};
+  var storagetransferMock = {
+    transferJobs: {
+      create: sinon.stub().callsArgWith(1, null, transferJobMock),
+      get: sinon.stub().callsArgWith(1, null, transferJobMock),
+      patch: sinon.stub().callsArgWith(1, null, transferJobMock),
+      list: sinon.stub().callsArgWith(1, null, { transferJobs: [transferJobMock] })
+    },
+    transferOperations: {
+      get: sinon.stub().callsArgWith(1, null, transferOperationMock),
+      pause: sinon.stub().callsArgWith(1, null, transferOperationMock),
+      resume: sinon.stub().callsArgWith(1, null, transferOperationMock),
+      list: sinon.stub().callsArgWith(1, null, { operations: [transferOperationMock] })
+    }
+  };
+  var googleapisMock = {
+    storagetransfer: sinon.stub().returns(storagetransferMock),
```

Comment:
I kept the "t" lowercase because that's how the api itself is named when you're building the client. `var storagetransfer = google.storagetransfer('v1');`